### ISSUE TITLE
[Question] Immutable 

### ISSUE THREAD

**Issue by aveiros** [immutables/immutables#1128]
This is more of a question.

Is there a way to define the behaviour of JsonSerialize and JsonDeserialize as part of a Custom Style (MyStyle)?  Consider the following example, I have to add JsonSerialize and JsonDeserialize (with builder name) to all the classes in order for Serialization and Deserialization for work in Jackson. Is there any way to this "dynamically" and part of the custom style?

```
@Value.Immutable
@MyStyle
@JsonSerialize
@JsonDeserialize(builder = ImmutableA.Builder.class)
public interface A {
     B getB();
}
```

```
@Value.Immutable
@MyStyle
@JsonSerialize
@JsonDeserialize(builder = ImmutableB.Builder.class)
public interface B {
     String getString();
}
```

**Comment by gdabski** [comment]
I don't think you need to specify `@JsonSerialize` on every [abstract value type](http://immutables.github.io/immutable.html#abstract-value-type). First, without any properties set as in your example it does nothing (it enables Jackson support in Immutables, but so does presence of `@JsonDeserialize`), and secondly you can anyway include it in your custom `@MyStyle` annotation.

As for dynamically specifying `@JsonDeserialize(builder = ...)` it is a valid feature request that I'd like to see myself. Right now, if you want to use the builder for deserialization, you must specify the concrete generated builder class on each abstract value type.
